### DISCUSSION TITLE
What is the proper way to preform backups?

### DISCUSSION THREAD

**Discussion by Hugo-Persson** [moghtech/komodo#216] [Q&A] [answered]
Hi!

What is the proper way to be able to recreate Komodo in case of disaster event such as server dies?

I know I can use resource sync and git repos for my actual stacks, but how do I properly back up Komodo core? Such as users, git providers etc?

**Comment by mbecker20** [comment, marked as answer]
You have to keep in mind a couple locations to backup:

1. **Config files**: Your compose.yaml, compose.env, core.config.toml for Komodo

2. The **Komodo database volume / folder**. For Mongo users, see https://www.mongodb.com/docs/manual/core/backups/#back-up-with-filesystem-snapshots. SQLite and Postgres users can refer to backup documentation for those databases. In general this is done most reliably using database dump, although file system copy usually works too. Before copy database files directly, you should stop the database for consistency. If you don't want to stop the database, database dump won't require this, but a little more work to script.

3. **Your Git Provider**. If you use Github it's not an issue, but if you are self hosting git provider like Gitea, make sure the files here are recover. Refer to your git provider for backup instructions.

4. If using "Files on Server" mode Stacks, the files just sit in a directory on your server, Komodo won't do anything to back them up. Make sure to back up the compose / config files on these servers.

5. Any data volumes / folders in use by your Deployments / Stacks. The data used by other applications managed by Komodo is still users responsibility to back up.

**Reply by Hugo-Persson** [reply]
Thank you for such great answer!

**Reply by Hugo-Persson** [reply]
Quick follow-up question @mbecker20, if I am using multiple servers, is it enough to back up database for the core and have server config in resource sync?

**Reply by mbecker20** [reply]
Komodo won't store any data on your other servers, the only persistent data for Komodo backup is the Core database, like you mentioned. On your other servers, just backup any data produced by applications you run with Komodo. The only other exception is compose files for Stacks in "Files on Server" that i mentioned before.

**Reply by Hugo-Persson** [reply]
Nice! Thank you so much for great answers and a great project! Really enjoying this over Portainer
